core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2105,6 +2120,23 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  // used as part of the workflow metadata permission check - more detail at workflowMetadata()
+
+  // require submission to be present, but don't require the workflow to reference it
+  // if the workflow does reference the submission, return its executionServiceKey
+
+  private def withSubmissionAndWorkflowExecutionServiceKey[T](workspaceContext: SlickWorkspaceContext, submissionId: String, workflowId: String, dataAccess: DataAccess)(op: (Option[String]) => ReadWriteAction[T]): ReadWriteAction[T] = {
+    withSubmission(workspaceContext, submissionId, dataAccess) { _ =>
```

Comment:
`withSubmission` is really heavy to use in this situation. I think all you're doing here is validating that a submission with that id exists, correct? If so, is there something more optimized you can use? IIRC `withSubmission` loads all the workflows and entities and such, which you don't need.

Reply:
Good point.  I'll optimize.

I bet if we decided to hunt for them we could find many other examples like this in Rawls.

Reply:
yeah, I'm sure there are a lot of existing cases of this - just don't want to add more!